File: medium/array-of-products/solution-1.js
array = [5, 1, 4, 2]

function arrayOfProducts(array) {
  const products = [];

  let i = 0;
  while (i < array.length) {
    const currNum = array.shift();
      console.log(currNum)
    const currentTotalProd = array.reduce((preVal, currVal) => preVal * currVal);
      console.log(currentTotalProd)
      console.log("-------------------------")
    products[i] = currentTotalProd;
    array.push(currNum);
    i++
  }
  return products;
}

arrayOfProducts(array)

File: medium/array-of-products/solution-1.py
def arrayOfProducts(array):
    products = [1 for _ in range(len(array))]

    leftPointer = 1
    for i in range(len(array)):
        products[i] = leftPointer
        leftPointer *= array[i]

    rightPointer = 1
    for i in reversed(range(len(array))):
        products[i] *= rightPointer
        rightPointer *= array[i]

    return products
